packages/flame/lib/src/components/component_notifier.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+/// A [ChangeNotifier] that notifies its listeners when a [Component] is
+/// added or removed, or updated.
```

Comment:
A bit unclear about the functionality of this class:

It looks like it represents a _collection_ of components of a particular type [T], and the user can subscribe to get notifications when new components of type T are mounted into the game, or removed from the game. The "update" part is a bit more vague - how does this work if all components are updated on every tick regardless?

Reply:
Maybe we can expand a little bit more on the updated part, I agree it is a little bit vague.

This update term has nothing to do with the component tick. But rather to the Notifiable component sending a signal that something changed.

Maybe follow the ChangeNofier api and just call this notfied?